data/frameSimulator/include/ExcaliburFrame.h

```diff
@@ -0,0 +1,50 @@
+#ifndef FRAMESIMULATOR_EXCALIBURFRAME_H
+#define FRAMESIMULATOR_EXCALIBURFRAME_H
+
+#include <vector>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "Packet.h"
+
+namespace FrameSimulator {
+
+    class ExcaliburFrameSimulatorPlugin;
+
+    typedef std::vector<Packet> PacketList;
+
+    /** ExcaliburFrame class
+     *
+     * An excalibur frame and its packets; defines the SOF and EOF markers
+     * for reading an excalibur frame from a pcap file
+     */
+
+    class ExcaliburFrame {
+
+        friend class ExcaliburFrameSimulatorPlugin;
```

Comment:
Why does this need to be a friend class?

Reply:
So that ExcaliburFrameSimulatorPlugin could use the private members of ExcaliburFrame (SOF_markers, EOF_markers etc.). I could add get/set methods if preferred ...